pkg/helpers/struct.go

```diff
@@ -208,50 +208,87 @@ func AsSlice(x *messages.ListValue) []interface{} {
 // When converting an int64 or uint64 to a NumberValue, numeric precision loss
 // is possible since they are stored as a float64.
 func NewValue(v interface{}) (*messages.Value, error) {
-	switch v := v.(type) {
-	case nil:
+
+	if v == nil {
 		return NewNullValue(), nil
-	case bool:
-		return NewBoolValue(v), nil
-	case int:
-		return NewNumberValue(float64(v)), nil
-	case int32:
-		return NewNumberValue(float64(v)), nil
-	case int64:
-		return NewNumberValue(float64(v)), nil
-	case uint:
-		return NewNumberValue(float64(v)), nil
-	case uint32:
-		return NewNumberValue(float64(v)), nil
-	case uint64:
-		return NewNumberValue(float64(v)), nil
-	case float32:
-		return NewNumberValue(float64(v)), nil
-	case float64:
-		return NewNumberValue(float64(v)), nil
-	case string:
-		if !utf8.ValidString(v) {
+	}
+
+	switch reflect.TypeOf(v).Kind() {
```

Comment:
Would it be faster to only use the `reflect` API in the `default` case of the switch statement, or in the specific cases where it is needed?

Reply:
Yah, that was mostly for consistency, but it's probably faster to only fallback where needed.